### PR TITLE
Update examples to use wait_for_participant

### PR DESCRIPTION
**Pull request by davidzhao**
drive-by: also updated default DeepGram model to nova-2-general

### CODE REVIEW

**Comment by theomonnom** [comment]
awesome, can you also s/will_synthesize_assistant_reply/before_llm_cb

**Review by theomonnom — APPROVED** [review]
awesome, thx!

File: examples/voice-assistant/README.md [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
+- `custom_pronunciation.py`: using the `before_tts_cb` hook to customize how TTS pronounces words
+- `simple_rag`: a simple RAG assistant that answers questions by querying a embeddings index
+
+The demo assistants use:
 
-Both assistants use:
-- Deepgram for Speech-to-text
-- OpenAI for LLM
-- Elevenlabs for Text-to-speech
+- DeepGram for Speech-to-text
```
Comment (theomonnom) [inline examples/voice-assistant/README.md:12]: nit
```suggestion
- Deepgram for Speech-to-text
```

Comment (davidzhao) [reply examples/voice-assistant/README.md:12]: oops.. shouldn't have changed it

**Pull request merged**